src/resmom/mom_main.c

```diff
@@ -3667,6 +3668,15 @@ static handler_ret_t
 set_jobdir_root(char *value)
 {
 	char	*cleaned_value;
+	char    *endptr;
+
+	endptr = value;
+	while ((!isspace((int)*endptr)) && *endptr)
+		endptr++;
+	if (*endptr != '\0') {
+		*endptr = '\0';
+		endptr++;
+	}
```

Comment:
What about using strchr(endptr, ' ') or strpbrk()?

Reply:
Not in this case, since I need to terminate the current string on the first non-space character found. For example, given '$jobdir /tmp share', endptr at this time will be pointing to the space (' ') after '/tmp', and I need that terminated with \0. '/tmp' is then passed on as the jobdir_root value.
I do see that if there are multiple spaces after '/tmp' and before 'shared', the current code does not move past the space to get to 'shared', so I'll fix that.

Reply:
I still don't see why the functions won't work.  They search for the space, and you want to replace the space with a null byte.  To handle the multiple space case, maybe strtok_r() is the right answer.  That handles the case when the token separator appears multiple times between tokens.

Reply:
Ok, I'll re-implement like this:
```
--- mom_main.c  2020-09-01 13:06:16.324785944 -0700
***************
*** 3668,3681 ****
  set_jobdir_root(char *value)
  {
        char    *cleaned_value;
        char    *endptr;
  
!       endptr = value;
!       while ((!isspace((int)*endptr)) && *endptr)
!               endptr++;
!       if (*endptr != '\0') {
                *endptr = '\0';
                endptr++;
        }
  
        log_event(PBSEVENT_SYSTEM, PBS_EVENTCLASS_SERVER,
--- 3668,3682 ----
  set_jobdir_root(char *value)
  {
        char    *cleaned_value;
+       char    *p;
        char    *endptr;
  
!       p = value;
!       if ((endptr = strchr(p, ' ')) != NULL) {
                *endptr = '\0';
                endptr++;
+       } else {
+               endptr = p;
        }
  
        log_event(PBSEVENT_SYSTEM, PBS_EVENTCLASS_SERVER,
***************
*** 3708,3713 ****
--- 3709,3716 ----
        strcpy(pbs_jobdir_root, cleaned_value);
        free(cleaned_value);
  
+       while (isspace((int)*endptr))
+               endptr++;
        if (*endptr != '\0') {
                if (strcmp(endptr, "shared") == 0)
                        pbs_jobdir_root_shared = TRUE;
```
-----
After the function processed the jobdir_root value, I'll move past any spaces to look for the optional 'shared' directive.

Reply:
That will work, but you could replace it all with the following two lines:
A = strtok_r(p, " ", &savep);
B = strtok_r(NULL, " ", &savep);
A will always return p regardless if there are any spaces.  B will be the first character after any number of spaces that separate the end of A and the start of B.  If there are no spaces, B will be NULL.

Reply:
Ok, will do this.